src/state_machine/coordinator/fire.rs

```diff
@@ -1110,9 +1110,13 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
             .cloned()
             .flat_map(|pn| pn.nonces)
             .collect::<Vec<PublicNonce>>();
-        let (_, R) = compute::intermediate(&self.message, &party_ids, &nonces);
 
-        R
+        let Some(group_key) = self.aggregate_public_key else {
+            return Err(Error::MissingAggregatePublicKey);
+        };
+        let (_, R) = compute::intermediate(&self.message, group_key, &party_ids, &nonces);
+
+        Ok(R)
```

Comment:
```suggestion
        let (_, aggregate_nonce) =
            compute::intermediate(&self.message, group_key, &party_ids, &nonces);

        Ok(aggregate_nonce)
```
Looks like sometimes in this PR we use `R` and sometimes we use `aggregate_nonce`. Which makes sense since we don't want to make unnecessary changes. I like small diffs. But if we're already changing these lines then we might as well make this change too. Since the diff is essentially unchanged.

Reply:
Done